manifests/resource/repository/docker/proxy.pp

```diff
@@ -0,0 +1,115 @@
+# @summary Resource to manage docker proxy repository
+#
+# @param proxy_remote_url
+#   Docker repository url like https://registry-1.docker.io.
+# @param ensure
+#   Define if the resource should be created/present or deleted/absent.
+# @param http_client_auto_block
+#   Auto-block outbound connections on the repository if remote peer is detected as unreachable/unresponsive.
+# @param http_client_blocked
+#   Block outbound connections on the repository.
+# @param negative_cache_enabled
+#   Cache responses for content not present in the proxied repository.
+# @param negative_cache_time_to_live
+#   How long to cache the fact that a file was not found in the repository (in minutes).
+# @param online
+#   Enable this repository in nexus repository manager that it can be used.
+# @param proxy_content_max_age
+#   Max age of content (packages).
+# @param proxy_metadata_max_age
+#   Max age of the repository metadata.
+# @param storage_blob_store_name
+#   The name of the blobstore inside of nexus repository manager to be used. We suggest to use a own blobstore for each
+#   defined repository.
+# @param storage_strict_content_type_validation
+#   Validate that all content uploaded to this repository is of a MIME type appropriate for the repository format.
+# @param storage_write_policy
+#   Controls if deployments of and updates to artifacts are allowed.
+# @param docker_v1_enabled
+#   Allow clients to use the V1 API to interact with this repository.
+# @param docker_force_basic_auth
+#   Allow anonymous docker pull ( Docker Bearer Token Realm required ).
+# @param docker_http_port
+#   Create an HTTP connector at specified port. Normally used if the server is behind a secure proxy.
+# @param docker_https_port
+#   Create an HTTPS connector at specified port. Normally used if the server is configured for https.
+# @param docker_proxy_index_type
+#   Docker index type. See https://help.sonatype.com/repomanager3/nexus-repository-administration/formats/docker-registry/proxy-repository-for-docker#ProxyRepositoryforDocker-ConfiguringaCorrectRemoteStorageandDockerIndexURLPair
+# @param docker_proxy_index_url
+#   If docker_proxy_index_type is CUSTOM you have to set the uri of the index api.
+#
+# @example
+#   nexus::repository::docker::proxy { 'docker-docker.io':
+#      proxy_remote_url => 'https://registry-1.docker.io',
+#   }
+#
+define nexus::resource::repository::docker::proxy (
+  Stdlib::HTTPSUrl $proxy_remote_url,
```

Comment:
Should we default this to the default docker registry?
I mean, are the so many alternatives?

Reply:
https://codefresh.io/docs/docs/integrations/docker-registries/

Podman also asks (i think on the first usage) which registry should be used as default. Not sure about docker but i guess it defaults to docker hub.